### ISSUE TITLE
Support Scala-specific field names

### ISSUE THREAD

**Issue by chipsenkbeil** [ensime/ensime-server#1506]
Not sure if I'd really call this a bug, but @dickwall has already encountered a Scala-specific debugging issue.

```scala
package com.example

object Hello {
  def main(args: Array[String]): Unit = {
    val h = new Hello("Rory", 5)
    h.sayHello()
  }
}

class Hello(name: String, times: Int) {
  def sayHello(): Unit = {
    val actualTimes = times * 2

    for (i <- 1 to actualTimes) // Breakpoint on this line
      println(s"Hello $name")
  }
}
```

In the above example, when reaching a breakpoint on the line `for (i <- 1 to actualTimes)`, the following variables are listed: `actualTimes` as the only local variable, `times` as a field, and `com$example$Hello$$name` as another field. The issue is that it appears that object fields have an odd name when using the JDI of `<class>$$<field name>`. This results in lookup via `DebugObjectField(threadId, name)` failing to find the field.

To remedy this, it looks like it is time to add the first Scala-specific profile to the debugger api. All this profile needs to do is inherit from the pure profile and override the field info method that returns the name of the field to chop off everything but the actual name of the field (after the `$$`). I need to open a debugger API issue to link to this.

On a side note, there is also an issue that accessing a field containing a primitive value doesn't seem to be possible with the current ensime protocol messages. The only field-related message is `DebugObjectField`, which contains an object id and a field name. This means primitive fields do not have a location to send back upon request (returns `None` [here](https://github.com/ensime/ensime-server/blob/1.0/core/src/main/scala/org/ensime/core/debug/DebugActor.scala#L210)).

Depends on https://github.com/ensime/scala-debugger/issues/257 to be completed.

**Comment by fommil** [comment]
We have methods to do the lookup of the fqn, are they not useful here?

**Comment by fommil** [comment]
@dickwall confirmed that this has always been there so removing from 1.0 unless there is a quick fix.

**Comment by chipsenkbeil** [comment]
I'll add a quick profile to fix this for now. If we see other weird quirks, the profile can be augmented. Shouldn't take more than an hour.

**Comment by chipsenkbeil** [comment]
Resolver didn't seem to work in the PR. Fixed via #1527.
